components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -339,24 +352,30 @@ protected void setSubject(AuthenticationContext context, Map<String, Object> jso
 
     protected void buildClaims(AuthenticationContext context, Map<String, Object> jsonObject)
             throws ApplicationAuthenticatorException {
+
+        boolean isUserClaimUriLocal = false;
+        boolean prefixClaimDialectUriWithClaims = false;
+        ClaimConfig claimConfig = getAuthenticatorClaimConfigurations(context);
+        String authenticatorClaimDialectUri = getClaimDialectURI();
+        if (readParametersFromAuthenticatorConfig() != null && readParametersFromAuthenticatorConfig().containsKey(
+                FacebookAuthenticatorConstants.PREFIE_CLAIM_DIALECT_URI)) {
+            prefixClaimDialectUriWithClaims = true;
+            authenticatorClaimDialectUri += "/";
```

Comment:
instead of this kind of appending
it is better to use a private method named "getEffectiveClaimUri(String claimdialectUri, String Claim, boolean shouldPrefix)"
isn't it?